src/lib/checkers.js

```diff
@@ -22,7 +22,7 @@ function findDemotions(configured, retrieved) {
 
 function findPromotions(configured, retrieved) {
   const promotions = Object.keys(configured).filter(
-    (member) => configured[member] === 'ADMIN' && retrieved[member].role === 'MEMBER'
+    (member) => configured[member] === 'ADMIN' && retrieved[member] && retrieved[member].role === 'MEMBER'
```

Comment:
I'm okay with the change and being more defensive, but what does this indicate? if there was no retrieved user is that a malformed toml? if so should it warn the user? Was it a user that should have been added but wasn't (or wasn't yet?) should it warn the user? if it was a user that should have been added, you need to make sure you do all the add/remove of users first and only once finished validate everything about the users.

Reply:
From the bug:

> It looks like it's erroring out on the fact that your rmarkins-godaddy doesn't yet exist in the org, so it doesn't exist in the retrieved object.

So that indicates its the latter from my comments and you need to do some things serially and then retrieve the users again if any modifications were made.

Reply:
for the case of the new user, if we just ignore this, then it will look like it did all the work, but wouldn't have actually added and promoted the user, right?

Reply:
New members will get added directly to their appropriate role, so no promotion is necessary. This is only for pre-existing members who have been manually demoted through the github.com interface, or who have been moved from member to admin in the config.

Reply:
ah okay. I'd suggest adding a comment in there somewhere about that (JSDoc?)